0pdd.rb

```diff
@@ -213,11 +213,24 @@
 
 get '/snapshot' do
   content_type 'text/xml'
-  repo = repo(repo_name(params[:name]))
-  repo.push
-  xml = repo.xml
-  xml.xpath('//processing-instruction("xml-stylesheet")').remove
-  xml.to_s
+  name = repo_name(params[:name])
+  master = params[:branch]
+  uri = "git@github.com:#{name}.git"
+  begin
+    repo = GitRepo.new(
+      uri: uri,
+      name: name,
+      id_rsa: settings.config['id_rsa'],
+      dir: settings.temp_dir,
+      master: master || 'master'
+    )
+    repo.push
+    xml = repo.xml
+    xml.xpath('//processing-instruction("xml-stylesheet")').remove
+    xml.to_s
+  rescue StandardError
+    error 400, "Could not get snapshot for #{name}"
```

Comment:
@mbao01 I still don't understand why we need this. You catch 'StandardError' which can mean anything and then make a decision that it meant that the repo is not found. Maybe the database connection is lost? Maybe out of memory? Why "not found"?

Reply:
@yegor256 No, you are right. The error I meant to catch here is `Exec:Error`. I updated this.